samples/nrf9160/cloud_client/src/main.c

```diff
@@ -55,11 +55,21 @@ static void cloud_update_work_fn(struct k_work *work)
 
 	struct cloud_msg msg = {
 		.qos = CLOUD_QOS_AT_MOST_ONCE,
-		.endpoint.type = CLOUD_EP_TOPIC_MSG,
 		.buf = CONFIG_CLOUD_MESSAGE,
 		.len = strlen(CONFIG_CLOUD_MESSAGE)
 	};
 
+	/* When using the nRF Cloud backend data is sent to the message topic.
+	 * This is in order to visualize the data in the web UI terminal.
+	 * For Azure IoT Hub and AWS IoT, messages are addressed directly to the
+	 * device twin (Azure) or device shadow (AWS).
+	 */
+	if (strcmp(CONFIG_CLOUD_BACKEND, "NRF_CLOUD") == 0) {
```

Comment:
Alternatively, `if (IS_ENABLED(CONFIG_NRF_CLOUD))`

Reply:
It may be that multiple backends are enabled at the same time and that another backend is actually selected by the ```CONFIG_CLOUD_BACKEND```. We would then use the wrong endpoint for that backend using the ```IS_ENABLED``` approach.